sdks/python/apache_beam/testing/metric_result_matchers.py

```diff
@@ -0,0 +1,261 @@
+    return True
+
+  def describe_to(self, description):
+    if self.namespace:
+      description.append_text(" namespace: ")
+      self.namespace.describe_to(description)
+    if self.name:
+      description.append_text(" name: ")
+      self.name.describe_to(description)
+    if self.step:
+      description.append_text(" step: ")
+      self.step.describe_to(description)
+    if self.label_key:
+      description.append_text(" label_key: ")
+      self.label_key.describe_to(description)
+    if self.label_value:
+      description.append_text(" label_value: ")
+      self.label_value.describe_to(description)
+    if self.attempted_value is not None:
+      description.append_text(" attempted_value: ")
+      self.attempted_value.describe_to(description)
+    if self.committed_value is not None:
+      description.append_text(" committed_value: ")
+      self.committed_value.describe_to(description)
+
+  def describe_mismatch(self, metric_result, mismatch_description):
+    mismatch_description.append_text("was").append_value(metric_result)
+
+
+class DistributionMatcher(BaseMatcher):
+  """A PyHamcrest matcher that validates counter distributions."""
+
+  @with_input_types(
+      sum_value=typing.Tuple[int, Matcher],
+      count_value=typing.Tuple[int, Matcher],
+      min_value=typing.Tuple[int, Matcher],
+      max_value=typing.Tuple[int, Matcher])
+  def __init__(self, sum_value, count_value, min_value, max_value):
+    self.sum_value = _matcher_or_equal_to(sum_value)
+    self.count_value = _matcher_or_equal_to(count_value)
+    self.min_value = _matcher_or_equal_to(min_value)
+    self.max_value = _matcher_or_equal_to(max_value)
+
+  def _matches(self, distribution_result):
+    if not isinstance(distribution_result, DistributionResult):
+      return False
+    if self.sum_value and not self.sum_value.matches(distribution_result.sum):
+      return False
+    if self.count_value and not self.count_value.matches(
+        distribution_result.count):
+      return False
+    if self.min_value and not self.min_value.matches(distribution_result.min):
+      return False
+    if self.max_value and not self.max_value.matches(distribution_result.max):
+      return False
+    return True
+
+  def describe_to(self, description):
+    if self.sum_value:
+      description.append_text(" sum_value: ")
+      self.sum_value.describe_to(description)
+    if self.count_value:
+      description.append_text(" count_value: ")
+      self.count_value.describe_to(description)
+    if self.min_value:
+      description.append_text(" min_value: ")
+      self.min_value.describe_to(description)
+    if self.max_value:
+      description.append_text(" max_value: ")
+      self.max_value.describe_to(description)
+
+  def describe_mismatch(self, distribution_result, mismatch_description):
+    mismatch_description.append_text("was").append_value(distribution_result)
+
+@with_input_types(typing.Tuple[str, Matcher])
+def has_step_name(step):
```

Comment:
It would seam more Pythonic to just expose MetricResultMatcher and let users set the right keywords. (The pattern here is needed in java because one doesn't have keyword arguments.)